src/main/logic/effects/Invincible.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Invincible.h
+ *
+ *  Created on: 9 Apr 2015
+ *      Author: Will Poynter
+ */
+
+#ifndef INVINCIBLE_H_
+#define INVINCIBLE_H_
+
+#include "IEffect.h"
+
+namespace gs {
+
+template<class T>
+class Invincible: public IEffect {
+private:
+	T* obj;
+	double* gameTime;
+	double expirey;
+	bool first;
+	bool expired;
+public:
+	Invincible(T* _obj, double* _gameTime, long int duration) :
+		obj(_obj),
+		gameTime(_gameTime),
+		first(true),
+		expired(false){
+		expirey = *_gameTime + duration;
+	}
+	virtual ~Invincible() {}
+
+	bool hasExpired() {
+		return expired;
+	}
+	void operator() () {
+		if (first) {
```

Comment:
This flag seems a big of a hack, it looks like we're calling this function multiple times but really we only want something to happen at the start and at the end